app/src/main/java/org/simple/clinic/bp/BloodPressureHistoryListItemDataSource.kt

```diff
@@ -19,12 +28,91 @@ class BloodPressureHistoryListItemDataSource(
     private val canEditFor: Duration,
     private val source: PositionalDataSource<BloodPressureMeasurement>
 ) : PositionalDataSource<BloodPressureHistoryListItem>() {
-  override fun loadRange(params: LoadRangeParams, callback: LoadRangeCallback<BloodPressureHistoryListItem>) {
 
+  private val invalidatedCallback = InvalidatedCallback { invalidate() }
+
+  init {
+    source.addInvalidatedCallback(invalidatedCallback)
+  }
+
+  override fun loadRange(params: LoadRangeParams, callback: LoadRangeCallback<BloodPressureHistoryListItem>) {
+    val newParams = LoadRangeParams(params.startPosition - 1, params.loadSize)
+    source.loadRange(newParams, object : LoadRangeCallback<BloodPressureMeasurement>() {
+      override fun onResult(measurements: MutableList<BloodPressureMeasurement>) {
+        val measurementListItems = convertToBloodPressureHistoryListItems(measurements)
+        val data = if (params.startPosition == 0) {
+          listOf(NewBpButton) + measurementListItems
+        } else {
+          measurementListItems
+        }
+        callback.onResult(data)
+      }
+    })
   }
 
   override fun loadInitial(params: LoadInitialParams, callback: LoadInitialCallback<BloodPressureHistoryListItem>) {
+    val newParams = LoadInitialParams(
+        params.requestedStartPosition,
```

Comment:
I initial used that, but since we have orientation locked I removed that. But re-adding it.

Reply:
I don't see this. Was it not pushed?

Reply:
Oh, I think I made a different change. What should happen if it's not zero? Should we set the start position to be  0?

Reply:
I think in this specific case, we will need to avoid changing the params start position since the position is not 0 anymore.

Let's punt this later since we currently do not have a screen which gets loaded after the BP history screen. We can revisit this when we add pagination support to one of the other screens.